spec/models/spree/product_spec.rb

```diff
@@ -11,10 +11,18 @@ module Spree
       context '#duplicate' do
         it 'duplicates product' do
           clone = product.duplicate
+
+          expect(clone).to be_persisted
           expect(clone.name).to eq 'COPY OF ' + product.name
           expect(clone.sku).to eq ""
           expect(clone.image).to eq product.image
         end
+
+        it 'fails to duplicate the product' do
+          pending 'test for all the use cases by which the +product.save+ would return false.'
```

Comment:
It's a very rare edge case because:
a) The product we will be cloning must already have been passed through the saving process, the `new_product.save` most probably will also pass as well.
b) The case in which it could fail I think would be for very old products i.e. Now in the present new validations might have been added in the model, and cloning those products might result in this case due to failing validations. So, in that case `.save` would return false. And that's what I'm assuming is the failure system spec for the product clone.

Maybe I'm missing something here, so in that case, I've created a pending spec, on which we can do futhur spike in a separate issue.
Please let me know what you think about it. Thanks.